### PR TITLE
Purifier carrier rework

### PR DESCRIPTION
**Pull request by MatthewMarinets**
Essentially redid / adapted the changes from PR #173.

* Renamed the unit to the Trireme
* Replaced the purifier interceptors with Bombers, which are larger and drop bombs which do big splash damage but only hit ground

Pairs with [client PR #299](https://github.com/Ziktofel/Archipelago/pull/299)

In some quick testing, I found the damage output vs ground was absolutely ridiculous, but it wiped out friendly units very readily, which I think is a fun balance. If we remove friendly-fire, I'd want to also reduce the damage output.

image
(Corrected this saying they could hit air after taking the picture):
image
image

### CODE REVIEW

**Comment by MatthewMarinets** [comment]
Took notes on what was added/removed:
[trireme.md](https://github.com/user-attachments/files/16885987/trireme.md)

**Comment by Ziktofel** [comment]
Damage would be subject to playtesting

**Comment by Ziktofel** [comment]
Snímek obrazovky pořízený 2024-09-05 23-21-50
If I attack an air target, the bomber interceptors move out, and stop near Trireme

**Comment by Ziktofel** [comment]
The thing is that before the unit had just real weapon that handled both interceptors and beam with both Ground and Air targeting, while the beam started to work only if the attack target is air. IIRC it's done this way (and not like multi-lock) due to effect being channeled, thus making the unit busy

**Comment by Ziktofel** [comment]
Use `Assets\Textures\btn-unit-collection-purifier-bomber.dds` asset to display bombers (dummy Trireme weapon, train icon)

**Comment by MatthewMarinets** [comment]
Fixed the icons, good catch.
image

Interceptors also no longer come out vs air.

**Comment by MatthewMarinets** [comment]
Missed the ability in that last one. Now fixed
image

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/EffectData.xml [hunk 1]
```diff
@@ -17644,6 +17632,84 @@
...
         <SearchFlags index="OffsetByUnitRadius" value="1"/>
     </CEffectEnumArea>
+    <CEffectCreatePersistent id="AP_ReleaseBombersPatrolCP">
+        <WhichLocation Value="CasterUnit"/>
+        <EditorCategories value="Race:Protoss"/>
+        <Flags index="RandomEffect" value="1"/>
+        <Flags index="RandomOffset" value="1"/>
+        <PeriodCount value="1"/>
+        <PeriodicEffectArray value="ReleaseBombersPatrolLeftTurnCP"/>
+        <PeriodicEffectArray value="ReleaseBombersPatrolRightTurnCP"/>
```
Comment (Ziktofel) [inline Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/EffectData.xml:17642]: These effects don't exist

Comment (MatthewMarinets) [reply Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/EffectData.xml]: Fixed, copied the effects over from starcoop gamedata. Didn't look too closely to see if interceptor movement effects were reusable, but that didn't seem to be the case at a glance.

**Comment by MatthewMarinets** [comment]
Did a few tests to make sure things weren't broken, and found that they were crazy popular. Three triremes could take down a base, would 4-volley a nexus, etc. I tuned the numbers down a bit, so a volley of bombs is 3 bombs instead of 4, and reduced the base damage from 12 -> 9. Tried again, they're still just as flashy, and probably still too strong.

**Pull request merged**